### ISSUE TITLE
[Home]: Able to add duplicate home

### ISSUE THREAD

**Issue by PayalChoksey17** [GeriLife/wellbeing#380]
Steps to Reproduce
1. Login to application
2. Click on Home
3. Click on Add Home - " Haley mouth"
4. Enter the details
6. Repeat step #3 and #4 
7. Notice, duplicate homes are created.

Screenshot :http://prntscr.com/njhk6b

**Comment by shailee-m** [comment]
@brylie it is not possible to make home names unique groupwise as mongo does not support composite keys. is it okay to make the home names unique overall?

**Comment by brylie** [comment]
We have use-cases where homes have duplicate names across the system. E.g. "Ward 1" could be used in multiple group homes. Duplicate home names will be allowed with the multi-tenancy application, so we should carefully consider whether to take any action here.

**Comment by shailee-m** [comment]
@brylie Agreed. Yet a group must have unique names so we will have to consider this while making the multitenancy app.

**Comment by brylie** [comment]
Can we use a pre-save hook to check for a duplicate home name in the same group when saving a home? E.g. in the AutoForm beforeSubmit or the equivalent?

**Comment by shailee-m** [comment]
> Can we use a pre-save hook to check for a duplicate home name in the same group when saving a home? E.g. in the AutoForm beforeSubmit or the equivalent?

Yes good for all cases. We will only have think about data import feature

**Comment by brylie** [comment]
There is also a collection hooks package, which may be more or less maintained.
